src/test/kotlin/no/nb/mlt/wls/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,49 @@
+package no.nb.mlt.wls.controller
+
+import com.ninjasquad.springmockk.MockkBean
+import io.mockk.every
+import no.nb.mlt.wls.EnableTestcontainers
+import no.nb.mlt.wls.core.data.Environment
+import no.nb.mlt.wls.core.data.HostName
+import no.nb.mlt.wls.core.data.Owner
+import no.nb.mlt.wls.core.data.Packaging
+import no.nb.mlt.wls.product.model.Product
+import no.nb.mlt.wls.product.payloads.toApiPayload
+import no.nb.mlt.wls.product.service.ProductService
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+
+@SpringBootTest
+@EnableTestcontainers
+@AutoConfigureMockMvc
+class ProductControllerTest() {
+    @MockkBean
+    lateinit var productService: ProductService
+
+    val testProduct =
+        Product(
+            hostName = HostName.AXIELL,
+            hostId = "mlt-2048",
+            productCategory = "BOOK",
+            description = "Ringenes Herre samling",
+            packaging = Packaging.BOX,
+            location = "SYNQ_WAREHOUSE",
+            quantity = 1.0,
+            preferredEnvironment = Environment.NONE,
+            owner = Owner.NB
+        )
+
+    @Test
+    fun saveProductTest() {
+        every {
+            productService.save(testProduct.toApiPayload())
+        } returns ResponseEntity(testProduct.toApiPayload(), HttpStatus.OK)
+
+        val result = productService.save(testProduct.toApiPayload())
+        assertEquals(HttpStatus.OK, result.statusCode)
```

Comment:
This doesn't really test anything, you just test that the mock returns what you told it to return. Should test using `@WebMvcTest` and do actual HTTP calls to endpoint.
Might also want to use the actual service and database to ensure that we have that part of the code working. We should probably mock the SynQ service to avoid calling SynQ or just mock response from `RestTemplate` inside the `SynqService` class (will have to use Autowire to inject the mocked `RestTemplate` or some other form of DI)
That way we can test much more cases and ensure that it's all handled correctly (see return codes from method in Swagger for test cases), along with Client Auth

Reply:
Sorry it should be `@WebFluxTest`, not `@WebMvcTest`